### ISSUE TITLE
sc2: SC2-compatible CMA-ES over multi-head linear policy weights

### ISSUE THREAD

**Issue by espenhk** [espenhk/gamer-ai#108]
The existing `CMAESPolicy` (`framework/policies.py`) evolves the flat weight vector of `WeightedLinearPolicy`, which has three linear heads (steer, accel, brake) mapped to TMNF's 15-dim observation. For SC2, the natural equivalent is a **multi-head linear policy** with separate weight matrices for fn_idx selection and spatial cell selection — each a linear map from obs to logits.

## Multi-head linear policy for SC2

```
weights = [W_fn: shape (6, obs_dim),  W_spatial: shape (9, obs_dim)]

fn_logits      = W_fn @ obs      → softmax → fn_idx
spatial_logits = W_spatial @ obs → softmax → cell_idx → (x, y)
```

The flat parameter vector `θ = [W_fn.flat | W_spatial.flat]` has dimension `(6 + 9) × obs_dim`:
- Minigame (13-dim obs): `θ ∈ ℝ^195`
- Ladder (21-dim obs): `θ ∈ ℝ^315`

This is larger than TMNF's ~63 dims but well within CMA-ES's practical range (typically up to ~1000 dims).

## Key differences from `CMAESPolicy`

| Aspect | TMNF `CMAESPolicy` | SC2 `CMAESPolicy` |
|---|---|---|
| Weight shape | `[steer(15), accel(15), brake(15)]` | `[W_fn(6×obs_dim), W_spatial(9×obs_dim)]` |
| Output | Continuous `[-1, 1]³` | Discrete fn_idx + cell_idx |
| θ dimension | ~63 | 195 (minigame) / 315 (ladder) |
| Available-actions masking | N/A | Applied in `act()`, not in CMA-ES update |

CMA-ES update mechanics (weighted recombination, CSA, rank-1 + rank-μ covariance update) are **unchanged** — only weight packing/unpacking and action selection differ.

## SC2MultiHeadLinearPolicy (shared individual type)

This class is also used by `SC2GeneticPolicy` (see related issue). Implement it once in `games/sc2/sc2_policies.py` so both evolutionary algorithms share it:

```python
class SC2MultiHeadLinearPolicy:
    def __init__(self, obs_dim, n_fns=6, n_cells=9): ...
    def act(self, obs, info) -> np.ndarray: ...       # returns 4-vector [fn_idx, x, y, queue]
    def flat_weights(self) -> np.ndarray: ...          # θ vector for CMA-ES
    def from_flat(self, theta): ...                    # unpack θ → W_fn, W_spatial
    def mutated(self, scale, share) -> 'SC2MultiHeadLinearPolicy': ...
    def save(self, path): ...                          # YAML serialisation
```

## Recommended hyperparams (`games/sc2/config/training_params.yaml`)

```yaml
policy_type: sc2_cmaes
policy_params:
  population_size: 30      # larger θ space → more offspring for coverage
  initial_sigma: 0.5       # larger init step for higher-dim space
  eval_episodes: 2         # average over 2 episodes per individual (SC2 is noisy)
```

## Tasks

- [ ] Implement `SC2MultiHeadLinearPolicy` in `games/sc2/sc2_policies.py` with `W_fn`, `W_spatial`, available-actions masking in `act()`, and YAML serialisation
- [ ] Implement `SC2CMAESPolicy` subclassing `CMAESPolicy`, overriding weight packing/unpacking (`flat_weights()` / `from_flat()`)
- [ ] Champion serialisation to YAML compatible with analytics and weight heatmap tools
- [ ] Register as `sc2_cmaes` in the policy factory
- [ ] Tests: θ dimension = `(6+9) × obs_dim`; champion YAML round-trips losslessly; masking produces valid fn_idx; CMA-ES σ adapts across generations

## Relevant files

- `framework/policies.py` — `CMAESPolicy`, `WeightedLinearPolicy` base classes
- `games/sc2/actions.py` — grid cell → (x, y) coordinates, fn_id constants
- `games/sc2/obs_spec.py` — obs dimension count per map type
- New: `games/sc2/sc2_policies.py`
